src/pip/_internal/utils/misc.py

```diff
@@ -897,6 +901,17 @@ def remove_auth_from_url(url):
     return surl
 
 
+def redact_password_from_url(url):
```

Comment:
I would add a docstring (forgot to say this and my None comment in the previous review).